packages/cli/src/tools/packageManager.js

```diff
@@ -36,33 +64,20 @@ export function setProjectDir(dir: string) {
 }
 
 export function install(packageNames: Array<string>, options?: Options) {
-  return shouldUseYarn(options)
-    ? executeCommand('yarn', ['add', ...packageNames], options)
-    : executeCommand(
-        'npm',
-        ['install', ...packageNames, '--save', '--save-exact'],
-        options,
-      );
+  return configurePackageManager(packageNames, options, 'add');
 }
 
 export function installDev(packageNames: Array<string>, options?: Options) {
-  return shouldUseYarn(options)
-    ? executeCommand('yarn', ['add', '-D', ...packageNames], options)
-    : executeCommand(
-        'npm',
-        ['install', ...packageNames, '--save-dev', '--save-exact'],
-        options,
-      );
+  return configurePackageManager(packageNames, options, 'addDev');
 }
 
 export function uninstall(packageNames: Array<string>, options?: Options) {
-  return shouldUseYarn(options)
-    ? executeCommand('yarn', ['remove', ...packageNames], options)
-    : executeCommand('npm', ['uninstall', ...packageNames, '--save'], options);
+  return configurePackageManager(packageNames, options, 'remove');
 }
 
 export function installAll(options?: Options) {
-  return shouldUseYarn(options)
-    ? executeCommand('yarn', ['install'], options)
-    : executeCommand('npm', ['install'], options);
+  const pm = shouldUseYarn(options) ? 'yarn' : 'npm';
+  const pmConfig = packageManagers[pm];
+
+  return executeCommand(pm, pmConfig.install, options);
```

Comment:
The respective method doesn't include the usage of `packageNames` and I guess having a separate implementation makes sense :thinking:

Reply:
packageNames may be an empty array, nothing bad about it